sdk/tests/sdk/whirlpools/whirlpool-impl.test.ts

```diff
@@ -495,4 +495,217 @@ describe("whirlpool-impl", () => {
+    );
+
+    const newOwnerPositionTokenAccount = await createAssociatedTokenAccount(
+      ctx.provider,
+      positionWithFees.mintKeypair.publicKey,
+      otherWallet.publicKey,
+      ctx.wallet.publicKey
+    );
+
+    await transfer(provider, walletPositionTokenAccount, newOwnerPositionTokenAccount, 1);
+
+    const pool = await client.getPool(whirlpoolPda.publicKey, true);
+    const position = await client.getPosition(positionWithFees.publicKey, true);
+    const positionData = position.getData();
+    const poolData = pool.getData();
+
+    const decreaseLiquidityQuote = decreaseLiquidityQuoteByLiquidity(
+      position.getData().liquidity,
+      Percentage.fromDecimal(new Decimal(0)),
+      position,
+      pool
+    );
+
+    const tickLowerArrayData = await ctx.fetcher.getTickArray(
+      positionWithFees.tickArrayLower,
+      true
+    );
+
+    const tickUpperArrayData = await ctx.fetcher.getTickArray(
+      positionWithFees.tickArrayUpper,
+      true
+    );
+
+    const tickLower = TickArrayUtil.getTickFromArray(
+      tickLowerArrayData!,
+      tickLowerIndex,
+      tickSpacing
+    );
+
+    const tickUpper = TickArrayUtil.getTickFromArray(
+      tickUpperArrayData!,
+      tickUpperIndex,
+      tickSpacing
+    );
+
+    const feesQuote = collectFeesQuote({
+      whirlpool: poolData,
+      position: positionData,
+      tickLower,
+      tickUpper,
+    });
+
+    const dWalletTokenBAccount = await deriveATA(otherWallet.publicKey, poolData.tokenMintB);
+    const rewardAccount0 = await deriveATA(otherWallet.publicKey, poolData.rewardInfos[0].mint);
+    const rewardAccount1 = await deriveATA(otherWallet.publicKey, poolData.rewardInfos[1].mint);
+    const rewardAccount2 = await deriveATA(otherWallet.publicKey, poolData.rewardInfos[2].mint);
+
+    const txs = await pool.closePosition(
+      positionWithFees.publicKey,
+      new Percentage(new u64(10), new u64(100)),
+      otherWallet.publicKey,
+      otherWallet.publicKey,
+      ctx.wallet.publicKey
+    );
+
+    let ataTx: TransactionBuilder | undefined;
+    let closeTx: TransactionBuilder;
+    if (txs.length === 1) {
+      closeTx = txs[0];
+    } else if (txs.length === 2) {
+      ataTx = txs[0];
+      closeTx = txs[1];
+    } else {
+      throw new Error(`Invalid length for txs ${txs}`);
+    }
+
+    const otherWalletBalanceBefore = await ctx.connection.getBalance(otherWallet.publicKey);
+    const positionAccountBalance = await ctx.connection.getBalance(positionWithFees.publicKey);
+
+    await ataTx?.buildAndExecute();
+    await closeTx.addSigner(otherWallet).buildAndExecute();
+
+    const otherWalletBalanceAfter = await ctx.connection.getBalance(otherWallet.publicKey);
+
+    const minAccountExempt = await ctx.fetcher.getAccountRentExempt();
+    const solReceived = otherWalletBalanceAfter - otherWalletBalanceBefore;
+
+    // TODO: Why is the reward quote only correct when data is collected after the tx is ran?
```

Comment:
I'm still investigating this one though. 

CollectRewardsQuote should return the correct value using the data prior to the collection of rewards, not after.